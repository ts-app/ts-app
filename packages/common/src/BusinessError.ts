/**
 * This interface represents recoverable errors.
 */
export interface BusinessError {
  /**
   * Business error message.
   */
  message: string
}
