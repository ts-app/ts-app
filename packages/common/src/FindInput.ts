/**
 * Generic input parameters to perform a search with cursor based paging.
 */
export interface FindInput {
  q?: string
  limit?: number
  cursor?: string
}
