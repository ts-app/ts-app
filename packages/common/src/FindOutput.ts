/**
 * Generic output from a search with cursor based paging.
 */
export interface FindOutput<T, U = string> {
  docs: T[]
  cursor: U
}
