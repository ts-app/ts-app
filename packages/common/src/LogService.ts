/**
 * A LogService is used to write log and error messages.
 */
export abstract class LogService {
  abstract log (message: string): void

  abstract error (message: string, error?: any): void

  abstract debug (message: string | object): void
}
