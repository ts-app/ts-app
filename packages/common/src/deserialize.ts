import * as LZString from 'lz-string'

export function deserialize (data: any, options?: {
  mode:
    'direct' |
    'decompress' |
    'decompressToUTF16' |
    'decompressToBase64' |
    'decompressToEncodedURIComponent'
}) {
  const reviver = (key: any, data: any) => {
    if (typeof data === 'string' && /^\d\d\d\d-\d\d-\d\dT\d\d:\d\d:\d\d.\d\d\dZ$/.test(data)) {
      return new Date(data)
    }
    return data
  }

  const mode = options && options.mode || 'direct'
  let val: any

  switch (mode) {
    case 'direct':
      val = data
      break
    case 'decompress':
      val = LZString.decompress(data)
      break
    case 'decompressToUTF16':
      val = LZString.decompressFromUTF16(data)
      break
    case 'decompressToBase64':
      val = LZString.decompressFromBase64(data)
      break
    case 'decompressToEncodedURIComponent':
      val = LZString.decompressFromEncodedURIComponent(data)
      break
    default:
      throw new Error(`Unknown deserialization mode [${mode}]`)
  }

  return JSON.parse(val, reviver)
}
