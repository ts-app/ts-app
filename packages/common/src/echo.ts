export const echo = (msg: string) => (`>>> ${msg}`)
