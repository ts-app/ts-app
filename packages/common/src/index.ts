export * from './BusinessError'
export * from './ConsoleLogService'
export * from './echo'
export * from './FindInput'
export * from './FindOutput'
export * from './LogService'
export * from './makeTimestampable'
export * from './ofBusinessError'
export * from './Timestampable'
export * from './User'
export * from './UserProfile'
