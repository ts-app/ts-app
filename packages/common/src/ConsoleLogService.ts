import { LogService } from './LogService'

/**
 * LogService implementation that writes messages to console.
 */
export class ConsoleLogService extends LogService {
  error (message: string, error?: any): void {
    if (error) {
      console.error(message)
    } else {
      console.error(message, error)
    }
  }

  log (message: string): void {
    console.log(message)
  }

  debug (message: string | object): void {
    console.debug(message)
  }
}
