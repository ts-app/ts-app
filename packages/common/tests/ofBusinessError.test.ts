import { ofBusinessError } from '../src'

describe('ofBusinessError', () => {
  it('works', done => {
    ofBusinessError('bob').subscribe(
      val => expect(val).toMatchSnapshot(),
      undefined,
      () => done()
    )
  })
})
