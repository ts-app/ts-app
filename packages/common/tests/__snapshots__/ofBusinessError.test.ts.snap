// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ofBusinessError works 1`] = `
Object {
  "error": Object {
    "message": "bob",
  },
}
`;
